dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/PhysicalPlacementTask.java

```diff
@@ -16,87 +16,177 @@
+            default:
+                list = ImmutableList.of();
         }
         final GetPhysicalPlacementForObjectsSpectraS3Response response = client
                 .getPhysicalPlacementForObjectsSpectraS3(
                         new GetPhysicalPlacementForObjectsSpectraS3Request(value.getBucketName(), list));
-        return response.getPhysicalPlacementResult();
+        if (response.getPhysicalPlacementResult().getPools() == null) {
+            response.getPhysicalPlacementResult().setPools(EmptyList.INSTANCE);
+        }
+        if (response.getPhysicalPlacementResult().getTapes() == null) {
+            response.getPhysicalPlacementResult().setTapes(EmptyList.INSTANCE);
+        }
+        if (response.getPhysicalPlacementResult().getDs3Targets() == null) {
+            response.getPhysicalPlacementResult().setDs3Targets(EmptyList.INSTANCE);
+        }
+        return new PhysicalPlacementModel(
+                buildPoolEntries(response),
+                buildReplicationEntries(response),
+                buildTapeEntries(response)
+        );
     }
 
+    private List<TapeEntry> buildTapeEntries(final GetPhysicalPlacementForObjectsSpectraS3Response response) {
+        return response.getPhysicalPlacementResult()
+                .getTapes()
+                .stream()
+                .map(this::getTapeEntry)
+                .collect(GuavaCollectors.immutableList());
+    }
 
-    private static class GetDirectoryObjects extends Task<ListBucketResult> {
-        final String bucketName, directoryFullName;
-        final Ds3Common ds3Common;
+    private TapeEntry getTapeEntry(final Tape tape) {
+        return new TapeEntry(
+                tape.getBarCode(),
+                tape.getSerialNumber(),
+                tape.getType(),
+                tape.getState().name(),
+                tape.getWriteProtected(),
+                FileSizeFormatKt.toByteRepresentation(tape.getAvailableRawCapacity()),
+                FileSizeFormatKt.toByteRepresentation(tape.getTotalRawCapacity() - tape.getAvailableRawCapacity()),
+                getPartitionName(tape.getPartitionId()),
+                dateTimeUtils.format(tape.getLastModified()),
+                tape.getEjectLabel() == null ? "" : tape.getEjectLabel(),
+                tape.getEjectLocation() == null ? "" : tape.getEjectLocation());
+    }
 
-        public GetDirectoryObjects(final String bucketName, final String directoryFullName, final Ds3Common ds3Common) {
-            this.bucketName = bucketName;
-            this.directoryFullName = directoryFullName;
-            this.ds3Common = ds3Common;
+    private String getPartitionName(final UUID id) {
+        try {
+            return ds3Common.getCurrentSession().getClient()
+                    .getTapePartitionSpectraS3(new GetTapePartitionSpectraS3Request(id.toString()))
+                    .getTapePartitionResult()
+                    .getName();
+        } catch (final IOException e) {
```

Comment:
This needs to have a LOG.error statement that logs the exception in addition to the loggingService call.

Reply:
This should be fixed too